tests/unit_tests/c_wrappers/job_runner/test_event_reporter.py

```diff
@@ -153,3 +156,57 @@ def test_report_inconsistent_events(unused_tcp_port):
             match=r"Illegal transition None -> \(MessageType<Finish>,\)",
         ):
             reporter.report(Finish())
+
+
+def test_report_with_failed_reporter_but_finished_jobs(unused_tcp_port):
+    mock_send_retry_time = 2
```

Comment:
Add a comment that what we test is that the reporter is not hanging forever (assuming the ert code does a similiar `join`) when the `send` deliberately fails.